dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentRelationshipsHelper.java

```diff
@@ -0,0 +1,186 @@
+import com.dotmarketing.util.json.JSONException;
+import com.dotmarketing.util.json.JSONObject;
+import com.dotmarketing.viewtools.content.util.ContentUtils;
+import com.liferay.portal.model.User;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides the utility methods that interact with content relationships in dotCMS. These methods
+ * are used by the Content Relationships REST end-point.
+ *
+ * @author Will Ezell
+ * @author Jose Castro
+ * @version 4.2
+ * @since Oct 11, 2017
+ */
+public class ContentRelationshipsHelper implements Serializable {
+
+    private static final long serialVersionUID = -2520775127453216350L;
+
+    private final LanguageAPI languageAPI = APILocator.getLanguageAPI();
+    private final ContentletAPI contentletAPI = APILocator.getContentletAPI();
+
+    private static final boolean RESPECT_FE_ROLES = Boolean.TRUE;
+
+    /**
+     * Private constructor
+     */
+    private ContentRelationshipsHelper() {
+
+    }
+
+    /**
+     * Provides a singleton instance of the {@link ContentRelationshipsHelper}
+     */
+    private static class SingletonHolder {
+        private static final ContentRelationshipsHelper INSTANCE = new ContentRelationshipsHelper();
+    }
+
+    /**
+     * Returns a singleton instance of this class.
+     *
+     * @return A single instance of this class.
+     */
+    public static ContentRelationshipsHelper getInstance() {
+        return ContentRelationshipsHelper.SingletonHolder.INSTANCE;
+    }
+
+    /**
+     * Returns a list of contents and its relationships according to a given set of parameters.
+     *
+     * @param request   The {@link HttpServletRequest} object.
+     * @param user      The {@link User} performing this action.
+     * @param paramsMap A Map of parameters that will define the search criteria.
+     * @return The list of associated contents.
+     */
+    public List<Contentlet> getRelatedContent(final HttpServletRequest request, final User user,
+                                              final Map<String, String> paramsMap) {
+        final String query = paramsMap.get(RESTParams.QUERY.getValue());
+        final String id = paramsMap.get(RESTParams.ID.getValue());
+        String orderBy = paramsMap.get(RESTParams.ORDERBY.getValue());
+        final String limitStr = paramsMap.get(RESTParams.LIMIT.getValue());
+        final String offsetStr = paramsMap.get(RESTParams.OFFSET.getValue());
+        final String inode = paramsMap.get(RESTParams.INODE.getValue());
+        orderBy = UtilMethods.isSet(orderBy) ? orderBy : "modDate desc";
+        long language = this.languageAPI.getDefaultLanguage().getId();
+
+        if (null != paramsMap.get(RESTParams.LANGUAGE.getValue())) {
+            try {
+                language = Long.parseLong(paramsMap.get(RESTParams.LANGUAGE.getValue()));
+            } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)